### ISSUE TITLE
#include <HTTPClient.h>

### ISSUE THREAD

**Issue by damien1213** [colagrosso/avviso#9]
Hello, I opened the exemple Ringtheredphone, then I have the #include < HTTPClient.h > in wrong color, and when I check I have this message : 
RingTheRedPhone.ino:27:24: fatal error: HTTPClient.h: No such file or directory

I really don't understand what's going on thanks !

**Comment by colagrosso** [comment]
Did you copy the `HTTPClient/` folder to your into your Arduino’s `libraries/` folder?

Avviso requires the HTTPClient library:

https://github.com/colagrosso/avviso/tree/master/Arduino/libraries/HTTPClient

**Comment by damien1213** [comment]
I try with many different way but I m not sure that I m doing right. I have to copy the  cpp. or .h ? In a file like bloc-note? thanks for answering quickly and sorry for my english '^^

**Comment by damien1213** [comment]
Or if you have skype to try to explain to me cause it's very complicate for a beginer like me, mostly in english :p I m kinda lost

here my skype if you think it's better : dada1213

**Comment by colagrosso** [comment]
You copy the whole `HTTClient` folder. The trick is to find your Arduino’s libraries/ folder. On my computer they are in the `~/Documents/Arduino/libraries` folder. Here is what should be there:

```
~/Documents/Arduino/libraries/Avviso/
~/Documents/Arduino/libraries/AvvisoUSB/
~/Documents/Arduino/libraries/EthernetDHCP/
~/Documents/Arduino/libraries/EthernetDNS/
~/Documents/Arduino/libraries/HTTPClient/
```

You'll know if it's working if you can see these libraries in the Arduino IDE. They will appear in the "Import Library" menu.

Have another look at the part of the README titled "Getting Started: I have an Arduino, but no Ethernet shield":

https://github.com/colagrosso/avviso#getting-started-i-have-an-arduino-but-no-ethernet-shield

which also links to here:

http://www.arduino.cc/en/Hacking/Libraries

**Comment by damien1213** [comment]
I ll look at this but I already have the Ethernet shield plug to my Arduino

**Comment by damien1213** [comment]
Still not working I really need that you put an eye on my computer, if you have teamviewer or skype it would be great

**Comment by colagrosso** [comment]
What is in your Arduino libraries folder? Does it look like this?

```
~/Documents/Arduino/libraries/Avviso/
~/Documents/Arduino/libraries/AvvisoUSB/
~/Documents/Arduino/libraries/EthernetDHCP/
~/Documents/Arduino/libraries/EthernetDNS/
~/Documents/Arduino/libraries/HTTPClient/
```

 What is in your "Sketch > Import Library..." menu in the Arduino IDE?

This guide might be easier to follow:

http://arduino.cc/en/guide/libraries

Can you follow those instructions to install any library?

> Still not working I really need that you put an eye on my computer, if you have teamviewer or skype it would be great

Sorry, no.

**Comment by damien1213** [comment]
I have 2 different folders : 
1. program files(x86)/arduino/libraries (here I have a lot more folders)
2. Documents/arduino/libraries here I have almost nothing 

The fact is that I followed couples tutorials to figure out how to get this work and now this is like a mess. I think finally I make it right with the HTTPClient but now I have a new error  : 

C:\Users\Damien\Documents\Arduino\libraries\HTTPClient/HTTPClient.h:27:33: fatal error: TCPSocketConnection.h: No such file or directory

Tomorrow I ll look at the link you gave me, cause I have the problem since 3 weeks and I spend all the afternoon again to figure it out without sucess. I m kinda tired to spend hours by doing nothing ^^
(and I m on windows, I don't know if it change something)

**Comment by damien1213** [comment]
I get it but when I check it I have now this : RingTheRedPhone.ino:29:20: fatal error: Avviso.h: No such file or directory

But I have the Avviso.h in :/

**Comment by colagrosso** [comment]
Sorry, I don't know why your installation doesn't work. My guess is that the files are in the wrong place.

You could try the "Importing a .zip Library" instructions here:

http://arduino.cc/en/guide/libraries#toc4

I haven't tried those instructions yet, but they look simpler than the manual process we have been trying.

**Comment by damien1213** [comment]
I finally get this work !!!! In fact I had 2 problems, the first one was silly :P I changed my port who was (COM4) to COM5 but didn't change the settings.
The seconde one was that I didn't put de Library correctly.. But now everthing work well, I can receive notification on NMA. I ll perform in my system now :) 

Thanks again for your help, you rock !

**Comment by colagrosso** [comment]
Great job. Way to go.

**Comment by nickmthw** [comment]
When I download the ZIP file from here..there are only library folders..
I don't see EthernetDCHP and EthernetDNS.
and I don't see them in the factory Arduino folder to copy them from there.
But the sketch seems to know where they are...as the color is changed like the rest.
I have An UNO and a Ethernet Shield...
Im guessing I used the correct sketch from the folder 
Avviso/examples/RingTheRedPhone

when I compile..i get this error below..
I have no idea what is wrong

Arduino: 1.6.4 (Windows 7), Board: "Arduino Uno"

In file included from C:\Users\cncdraw\Documents\Arduino\libraries\HTTPClient\HTTPClient.cpp:27:0:
C:\Users\cncdraw\Documents\Arduino\libraries\HTTPClient\HTTPClient.cpp: In member function '__file\* HTTPClient::getURI(char_, http_client_parameter_, http_client_parameter_)':
C:\Users\cncdraw\Documents\Arduino\libraries\HTTPClient\HTTPClient.cpp:78:45: error: invalid conversion from 'const char_' to 'char_' [-fpermissive]
   sendUriAndHeaders(result, this->hostName, PSTR("GET"), uri, parameters,
                                             ^
In file included from C:\Users\cncdraw\Documents\Arduino\libraries\HTTPClient\HTTPClient.cpp:29:0:
C:\Users\cncdraw\Documents\Arduino\libraries\HTTPClient\HTTPClient.h:191:3: error:   initializing argument 3 of 'char HTTPClient::sendUriAndHeaders(__file_, char_, char_, char_, http_client_parameter_, http_client_parameter_)' [-fpermissive]
   sendUriAndHeaders(FILE_ stream, char\* hostName, char\* requestType, char\* uri,
   ^
In file included from C:\Users\cncdraw\Documents\Arduino\libraries\HTTPClient\HTTPClient.cpp:27:0:
C:\Users\cncdraw\Documents\Arduino\libraries\HTTPClient\HTTPClient.cpp: In member function '__file\* HTTPClient::postURI(char_, http_client_parameter_, char_, http_client_parameter_)':
C:\Users\cncdraw\Documents\Arduino\libraries\HTTPClient\HTTPClient.cpp:107:45: error: invalid conversion from 'const char_' to 'char_' [-fpermissive]
   sendUriAndHeaders(result, this->hostName, PSTR("POST"), uri, parameters,
                                             ^
In file included from C:\Users\cncdraw\Documents\Arduino\libraries\HTTPClient\HTTPClient.cpp:29:0:
C:\Users\cncdraw\Documents\Arduino\libraries\HTTPClient\HTTPClient.h:191:3: error:   initializing argument 3 of 'char HTTPClient::sendUriAndHeaders(__file_, char_, char_, char_, http_client_parameter_, http_client_parameter_)' [-fpermissive]
   sendUriAndHeaders(FILE\* stream, char\* hostName, char\* requestType, char\* uri,
   ^
In file included from C:\Users\cncdraw\Documents\Arduino\libraries\HTTPClient\HTTPClient.cpp:27:0:
C:\Users\cncdraw\Documents\Arduino\libraries\HTTPClient\HTTPClient.cpp: In member function '__file\* HTTPClient::putURI(char_, http_client_parameter_, char_, http_client_parameter_)':
C:\Users\cncdraw\Documents\Arduino\libraries\HTTPClient\HTTPClient.cpp:134:45: error: invalid conversion from 'const char_' to 'char_' [-fpermissive]
   sendUriAndHeaders(result, this->hostName, PSTR("PUT"), uri, parameters,
                                             ^
In file included from C:\Users\cncdraw\Documents\Arduino\libraries\HTTPClient\HTTPClient.cpp:29:0:
C:\Users\cncdraw\Documents\Arduino\libraries\HTTPClient\HTTPClient.h:191:3: error:   initializing argument 3 of 'char HTTPClient::sendUriAndHeaders(__file_, char_, char_, char_, http_client_parameter_, http_client_parameter_)' [-fpermissive]
   sendUriAndHeaders(FILE\* stream, char\* hostName, char\* requestType, char\* uri,
   ^
Error compiling.

  This report would have more information with
  "Show verbose output during compilation"
  enabled in File > Preferences.

**Comment by nickmthw** [comment]
![compile error](https://cloud.githubusercontent.com/assets/13433838/8805459/59539332-2f9f-11e5-9ccf-83450bfa64f7.jpg)

**Comment by colagrosso** [comment]
Thanks for pasting the error message and the screenshot, nickmthw. It looks like the error is during the compiling of HTTPClient. Avviso includes a copy of HTTPClient that I got from here:

https://github.com/interactive-matter/HTTPClient

Can you try downloading that library and trying to compile one if its example sketches?

**Comment by nickmthw** [comment]
So add this to my Library and try to compile the same sketch again?

**Comment by nickmthw** [comment]
Ok.. took out the HTTPclient that came in the Avviso-master download
moved it up one folder in the arduino sketch folder
and in the libraries folder..i used the HTTPclient-master
I just downloaded and it works..or it compiles anyway
when I get home ..then I will upload it to the shield and see if it work
what Im trying to do..is ..if a pin is at a certain condition..it will trigger a push alert
lets say..a photocell sees light and voltage is read..it will alert me someone has opened the door in my shop..or  something like that

**Comment by nickmthw** [comment]
ok..
I have a question
I am trying to use your code for Push alerts in my shop..so I have setup a sample to test
What I have it..
If SigPin reads HIGH.
Then send me a push notification
and digitalWrite to OutPin and turn an LED.

My router is to far away from my Laptop.. so I Couldn't use the IDE and the arduino at same time
to open the serial port and see what was happening.

Does this code look like it will do as I want?

#include <SPI.h>
#include <Ethernet.h>

#include <HTTPClient.h>
#include <Ethernet.h>
#include <Avviso.h>

#define DEBUG 1

byte mac[] = { 0xDE, 0xAD, 0xBE, 0xEF, 0xFE, 0xED };

int counterValue;
int SigPin=11;
int OutPin=8;

void setup() {
  counterValue = 10;
  Serial.begin(115200);
  if (DEBUG) Serial.println("Attempting to obtain a DHCP lease...");
  Ethernet.begin(mac);

  if (DEBUG) {
    Serial.print("My IP address is: ");
    Ethernet.localIP().printTo(Serial);
    Serial.println();

```
Serial.print("Gateway IP address is ");
Ethernet.gatewayIP().printTo(Serial);
Serial.println();

Serial.print("DNS IP address is ");
Ethernet.dnsServerIP().printTo(Serial);
Serial.println();
pinMode(SigPin,INPUT);
pinMode(OutPin,OUTPUT);
```

  }

  Avviso.begin();

  Avviso.setApiKey("e6b175cac095e82b01e6050ecff22eba60f7d15c");

  Avviso.setApplicationName("ALERT");
}

void loop(){
  int Volts = digitalRead(SigPin);  

if (Volts,HIGH){
  Avviso.push("Hey", "Some one is here",0);//need 3 events here
  delay (500);
  digitalWrite(OutPin,HIGH);
  delay(500);
  digitalWrite(OutPin,LOW);
  delay(500);
}
else{
  digitalWrite(OutPin,LOW);
}
  }

**Comment by nickmthw** [comment]
![push 1](https://cloud.githubusercontent.com/assets/13433838/8825000/4ef79cbc-3049-11e5-9044-8c02740c9672.jpg)
![push 2](https://cloud.githubusercontent.com/assets/13433838/8825002/500a5f90-3049-11e5-8fd9-bdd3cce20481.jpg)

**Comment by nickmthw** [comment]
The event name and description is just something I wrote that would notify my
someone is at my shop
it doesn't have to anything specific does it?

**Comment by colagrosso** [comment]
Code looks fine at a first pass. Yes, the event name and description are free for you to set.

**Comment by nickmthw** [comment]
Ok...code worked as far as sending goes...BUT..it keeps sending with out the "Front input" having voltage on it. Its like it needs something after the "IF" statement to say
doesn't do anything  "IF Front is NOT HIGH"
And I can't figure what do add to make it NOT send if Front is NOT high

It just keeps sending and sending and sending
Like how ever fast the controller scans..it sends
20-30 in like 3-4 seconds it sent me

Here is my new code..

**Comment by nickmthw** [comment]
#include <SPI.h>
#include <Ethernet.h>
#include <HTTPClient.h>
#include <Ethernet.h>
#include <Avviso.h>
#define DEBUG 1
byte mac[] = { 0xDE, 0xAD, 0xBE, 0xEF, 0xFE, 0xED };

int counterValue;      //maybe used for a counter if needed
int Front=11;        //front door switch
int Fopen=10;       //front door open output if i add a light
int Window=9;      //window switch
int Wopen=8;      //window open output light if i add add a light

void setup() {
  counterValue = 10;  
pinMode(Front,INPUT);      //made front switch an input
pinMode(Fopen,OUTPUT);      //made front output for light
pinMode(Window,INPUT);      //made window switch an input
pinMode(Wopen,OUTPUT);      //made front output for light  

  Serial.begin(115200);                // open serial port
  if (DEBUG) Serial.println("Attempting to obtain a DHCP lease...");
  Ethernet.begin(mac);                //all this below gathers network info and prints is
  if (DEBUG) {
    Serial.print("My IP address is: ");
    Ethernet.localIP().printTo(Serial);
    Serial.println();
    Serial.print("Gateway IP address is ");
    Ethernet.gatewayIP().printTo(Serial);
    Serial.println();
    Serial.print("DNS IP address is ");
    Ethernet.dnsServerIP().printTo(Serial);
    Serial.println();
  }
   Avviso.begin();                           //begin the library for Avviso app
   Avviso.setApiKey("e6b175cac095e82b01e6050ecff22eba60f7d15c");//phone receiver key
   Avviso.setApplicationName("ALERT");      //app name saying alert
}

void loop() {
  int Frontopen = digitalRead (Front);    //reading front door switch to see if open
  int Windowopen =digitalRead (Window);  //reading window switch to see if open

  if(Frontopen=HIGH){                           //Check to see if Front door is open  
    Avviso.push("Hey", "Front door is open",1);//if it is..send message and priority
    delay(1500);  
}
  else{                                  // do nothing if door isn't opened...i hope

}

  if(Windowopen==HIGH){                //Check to see IF..Window is open
    Avviso.push("Hey", "Window is open",1);//send message if window is open
    delay(1500);

}
  else{

}
}

**Comment by nickmthw** [comment]
![mid of code](https://cloud.githubusercontent.com/assets/13433838/8864071/ea5f4f1a-316b-11e5-85dd-76227ff8a7c6.jpg)
![Uploading top of code.jpg…]()
![Uploading bottom of code.jpg…]()

**Comment by nickmthw** [comment]
![top of code](https://cloud.githubusercontent.com/assets/13433838/8864096/23463e7e-316c-11e5-95d2-0a5bd28ec8b5.jpg)
![mid of code](https://cloud.githubusercontent.com/assets/13433838/8864099/27bc20a4-316c-11e5-8129-a40149ff1359.jpg)
![bottom of code](https://cloud.githubusercontent.com/assets/13433838/8864100/2aa52586-316c-11e5-8f92-f426a0336697.jpg)

**Comment by nickmthw** [comment]
Got it now..

Thanks

**Comment by ds200912** [comment]
Hi,
I got the ring the red phone to work when  I didn't have the Arduino connected. I followed the steps under having an Arduino without an ethernet shield and couldn't get the notification to work. So today I bought an Ethernet Shield, and followed the steps under that section and this printed at the bottom of the screen "Sending push notification...Error. Server returned: 201". Any help would be greatly appreciated, I'm doing a project I have to be finished with tomorrow. Thanks
